File: src/main/java/ru/ssau/tourism/entities/TouristInfo.java
package ru.ssau.tourism.entities;

import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Entity
@Setter
@NoArgsConstructor
@EqualsAndHashCode(of = {"id"})
public class TouristInfo {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @OneToOne
    @JoinColumn(name = "tourist_id")
    private Tourist tourist;
    @Column
    private String passport;
    @Column
    private String city;
    @Column
    private String country;
    @Column
    private Integer phone_number;
    @Column
    private Integer index;

    public Long getId() {
        return id;
    }

    public Tourist getTourist() {
        return tourist;
    }

    public String getPassport() {
        return passport;
    }

    public String getCity() {
        return city;
    }

    public String getCountry() {
        return country;
    }

    public Integer getPhone_number() {
        return phone_number;
    }

    public Integer getIndex() {
        return index;
    }
}

File: src/main/kotlin/ru/ssau/tourism/entities/TouristInfo.kt
package ru.ssau.tourism.entities

import javax.persistence.*

@Entity
class TouristInfo constructor(
		@Id
		@GeneratedValue(strategy = GenerationType.AUTO)
		var id: Long? = null,
		@OneToOne
		@JoinColumn(name = "tourist_id")
		var tourist: Tourist? = null,
		@Column
		var passport: String? = null,
		@Column
		var city: String? = null,
		@Column
		var country: String? = null,
		@Column
		var phone_number: Int? = null,
		@Column
		var index: Int? = null
) {
	override fun equals(other: Any?): Boolean {
		if (this === other) return true
		if (other?.javaClass != javaClass) return false
		other as TouristInfo
		if (id != other.id) return false
		if (passport != other.passport) return false
		return true
	}

	override fun hashCode(): Int {
		var result = id?.hashCode() ?: 0
		result = 31 * result + (passport?.hashCode() ?: 0)
		return result
	}
}
